### PR TITLE
[FIX] sale_mrp,stock : correctly split quantity between kit in package 

### PR DESCRIPTION
**Pull request by robinengels**
Current behavior:
When 2 steps delivery and product packing is activated, and have 2 kit products that have common components.
If you create a sale order with those 2 kits and pack them in the first step of the delivery. And in the second delivery step mark the pack as done the done quantity in the package are not correct.

Before this fix, all the products quantities from different kits were on the same stock move line and the other lines were ignored,
this result in lines with too much products and lines with no product. When validating this
incorrect transfer an unnecessary backorder was created and the sale order lines were not correctly marked as delivered.

Steps to reproduce:
- Activate 2 steps delivery and product packing.
- Create Kit A with Component A and Component B
- Create Kit B with Component A and Component B
- Create a SO with 1 Kit A and 1 Kit B
- Confirm the SO
- Go in the first step of the delivery and put everything in a pack
- Go in the second step of the delivery and set the pack as done
- Go in the package details : The first component A has 1 reserved and 2 done, same for component B

opw-2754106
--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/stock/models/stock_package_level.py [hunk 1]
```diff
@@ -54,9 +55,14 @@ def _set_is_done(self):
                 if not package_level.is_fresh_package:
                     ml_update_dict = defaultdict(float)
                     for quant in package_level.package_id.quant_ids:
-                        corresponding_ml = package_level.move_line_ids.filtered(lambda ml: ml.product_id == quant.product_id and ml.lot_id == quant.lot_id)
-                        if corresponding_ml:
-                            ml_update_dict[corresponding_ml[0]] += quant.quantity
+                        corresponding_mls = package_level.move_line_ids.filtered(lambda ml: ml.product_id == quant.product_id and ml.lot_id == quant.lot_id)
+                        to_dispatch = quant.quantity
+                        if corresponding_mls:
+                            while float_compare(to_dispatch, 0, precision_digits=2) and corresponding_mls:
```
Comment (fmdl) [inline addons/stock/models/stock_package_level.py:61]: `float_compare(to_dispatch, 0, precision_digits=2)` --> `not float_is_zero(to_dispatch, precision_rounding=quant.product_uom_id.rounding)`

maybe use the precision rounding of the uom (and not 2 by default)

**Comment by robinengels** [comment]
@adwid

**Review by adwid — COMMENTED** [review]
It's a shame: quite overkill, but I guess we can't do anything else

`stock.quant` uses the product's UoM:
https://github.com/odoo/odoo/blob/b22a23435a0cc42b7084ded21dbd4ef3a4a2113e/addons/stock/models/stock_quant.py#L60-L62
so I think you should use `product_qty` on the SM, instead of `product_uom_qty`
https://github.com/odoo/odoo/blob/30cc4b5050f538ac4b9152dd78686682d42e7036/addons/stock/models/stock_move.py#L285-L287

(Could you rename your commit header ? it does not follow the guidelines https://www.odoo.com/documentation/15.0/developer/misc/other/guidelines.html#commit-message-header)

**Comment by Whenrow** [comment]
@robinengels Thank you
robodoo r+

**Pull request closed without merging**